sentry_sdk/integrations/wsgi.py

```diff
@@ -92,6 +94,16 @@ def get_request_url(environ, use_x_forwarded_for=False):
         wsgi_decoding_dance(environ.get("PATH_INFO") or "").lstrip("/"),
     )
 
+@contextmanager
+def conditional_context_manager(manager, condition):
```

Comment:
as a readable code thumbrule, make generalized abstractions only if necessary and try to articulate things closer to the actual logic.

Here, I'd prefer a specific profiler context manager in `profiler.py` like [this sessions one](https://github.com/getsentry/sentry-python/blob/68d77e3b6649f60f465d580f53307a129685a871/sentry_sdk/sessions.py#L37-L50).
```python
@contextmanager
def profiling(transaction, hub=None):
    # type: (Transaction, Optional[sentry_sdk.Hub]) -> Generator[None, None, None]
    if hub is None:
        hub = sentry_sdk.Hub.current
    if hub.client.options["enable_profiling"]:
        with Sampler(transaction):
            yield
    else:
        yield
```